Napzak-iOS/Napzak-iOS/Global/Components/NZSegmentedControl.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  NZSegmentedControl.swift
+//  Napzak-iOS
+//
+//  Created by 조혜린 on 4/15/25.
+//
+
+import SwiftUI
+
+struct NZSegmentedControl: View {
+    
+    //MARK: - Property Wrappers
+    
+    @Binding var selectedTabIndex: Int
+    
+    //MARK: - Properties
+    
+    let tabs: [String]
+    let spacing: CGFloat
+
+    private var columns: [GridItem] {
+        Array(repeating: GridItem(.flexible(), spacing: spacing), count: tabs.count)
+    }
+    
+    //MARK: - Main Body
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            segmentedButtons
+                .padding(.vertical, 5)
+            segmentedHighlighter
+        }
+    }
+}
+
+extension NZSegmentedControl {
+    
+    //MARK: - UI Properties
+    
+    private var segmentedButtons: some View {
+        LazyVGrid(columns: columns) {
+            ForEach(tabs.indices, id: \.self) { i in
+                Button {
+                    selectedTabIndex = i
+                } label: {
+                    Text(tabs[i])
+                        .applyNapzakFont(selectedTabIndex == i ? .body1Bold16 : .body2SemiBold16)
+                        .foregroundStyle(selectedTabIndex == i ? Color.napzakPrimary(.purple500): Color.napzakGrayScale(.gray200))
+                }
+                .frame(maxWidth: .infinity)
+                .frame(height: 36)
+            }
+        }
+    }
+    
+    private var segmentedHighlighter: some View {
+        LazyVGrid(columns: columns) {
+            ForEach(tabs.indices, id: \.self) { i in
+                Color.napzakPrimary(.purple500).opacity(selectedTabIndex == i ? 1 : 0)
+                    .frame(height: 2)
+            }
+        }
+    }
```

Comment:
```suggestion
    
       private var segmentedHighlighter: some View {
        GeometryReader { geo in
            let totalWidth = geo.size.width
            let tabWidth = totalWidth / CGFloat(tabs.count)
            
            Rectangle()
                .fill(Color.napzakPrimary(.purple500))
                .frame(width: tabWidth, height: 2)
                .offset(x: tabWidth * CGFloat(selectedTabIndex), y: 0)
                .animation(.easeInOut(duration: 0.3), value: selectedTabIndex)
        }
        .frame(height: 2)
    }
```
명세서에는 없긴 한데, 세그먼트 컨트롤하면 저는 하이라이트 바가 움직이는게 기본적으로 떠올라서 애니메이션 추가해봤어요.
나중에 애니메이션이 필요하다면 저는 위와 같은 방법을 적용할 것 같아요 
gif여서 실제로 더 부드럽게 움직입니다
![animation](https://github.com/user-attachments/assets/f4caf162-5276-409f-8b8c-526c9d83d57f)

Reply:
추후 스와이프 전환 등의 기능도 같이 기디와 논의해서 적용해보면 좋을 것 같아요~! 지금은 작업량이 많아질 것 같으니.. 우선 넘어가도록 할게요 좋은 의견 감사감사